### PR TITLE
Testing Tidy and Integration/DBMS split

### PR DESCRIPTION
**Pull request by cjllanwarne**

### CODE REVIEW

**Comment by cjllanwarne** [comment]
Especially interested in the opinions of @geoffjentry, @kshakir and @scottfrazer as people who seem to ken `sbt` better than I

File: project/Testing.scala [hunk 1]
```diff
@@ -8,31 +8,44 @@ object Testing {
...
   val testSettings = List(
-    // `test` (or `assembly`) - Run all tests, except docker and integration
+    // `test` (or `assembly`) - Run all tests, except docker and integration and DBMS
     testOptions in Test ++= defaultTestArgs,
     // `alltests:test` - Run all tests
     testOptions in AllTests := (testOptions in Test).value.diff(defaultTestArgs),
-    // `docker:test` - Run docker tests, except integration
-    testOptions in DockerTest := (testOptions in Test).value.diff(defaultTestArgs) ++
-      Seq(Tests.Argument("-n", "DockerTest"), Tests.Argument("-l", "CromwellIntegrationTest")),
+    // `notests:test` - Run no tests
```
Comment (scottfrazer) [inline project/Testing.scala]: I can't remember if you said you were going to change/remove this because `notests:test` is confusing...

Comment (geoffjentry) [inline project/Testing.scala]: Yeah - @cjllanwarne if you don't feel like removing it I'll do it shortly after on a followup PR

**Comment by geoffjentry** [comment]
:+1:

File: project/Testing.scala [hunk 2]
```diff
@@ -8,31 +8,44 @@ object Testing {
...
-    testOptions in CromwellIntegrationTest := (testOptions in Test).value.diff(defaultTestArgs) ++
-      Seq(Tests.Argument("-l", "DockerTest"), Tests.Argument("-n", "CromwellIntegrationTest")),
+    testOptions in NoDockerTest := (testOptions in AllTests).value ++ Seq(DontUseDockerTaggedTests),
+    // `integration:test` - Run only integration tests
+    testOptions in CromwellIntegrationTest := (testOptions in AllTests).value ++ Seq(UseCromwellIntegrationTaggedTests),
     // `nointegration:test` - Run all tests, except integration
-    testOptions in CromwellNoIntegrationTest := (testOptions in Test).value.diff(defaultTestArgs) ++
-      Seq(Tests.Argument("-l", "CromwellIntegrationTest"))
+    testOptions in CromwellNoIntegrationTest := (testOptions in AllTests).value ++ Seq(DontUseCromwellIntegrationTaggedTests),
+    // `dbms:test` - Run database management tests.
```
Comment (scottfrazer) [inline project/Testing.scala]: Something is confusing me here... I tried running `sbt test` and `sbt dbms:test` and it appears that it runs the same number of tests (263)

File: engine/src/test/scala/engine/db/slick/SlickDataAccessSpec.scala [hunk 3]
```diff
@@ -6,7 +6,7 @@ import java.util.UUID
 import better.files._
 import com.google.api.client.util.ExponentialBackOff
 import com.typesafe.config.ConfigFactory
-import cromwell.CromwellSpec.IntegrationTest
+import cromwell.CromwellSpec.{DbmsTest, IntegrationTest}
```
Comment (kshakir) [inline engine/src/test/scala/engine/db/slick/SlickDataAccessSpec.scala]: There's still a test tagged as `IntegrationTest`. Delete this import and intellij/sbt shall reveal the line.

File: project/Testing.scala [hunk 4]
```diff
@@ -8,31 +8,44 @@ object Testing {
...
+    testOptions in DbmsTest := (testOptions in Test).value ++ Seq(UseDbmsTaggedTests)
```
Comment (kshakir) [inline project/Testing.scala]: Related to Scott's note above, this should be `(testOptions in AllTests)`.

File: build.sbt [hunk 5]
```diff
@@ -16,6 +16,7 @@ lazy val engine = (project in file("engine"))
   .configs(NoDockerTest).settings(inConfig(NoDockerTest)(Defaults.testTasks): _*)
   .configs(CromwellIntegrationTest).settings(inConfig(CromwellIntegrationTest)(Defaults.testTasks): _*)
   .configs(CromwellNoIntegrationTest).settings(inConfig(CromwellNoIntegrationTest)(Defaults.testTasks): _*)
+  .configs(DbmsTest).settings(inConfig(CromwellIntegrationTest)(Defaults.testTasks): _*)
```
Comment (kshakir) [inline build.sbt]: `inConfig(DbmsTest)` (1 of 2)

File: build.sbt [hunk 6]
```diff
@@ -27,5 +28,6 @@ lazy val root = (project in file("."))
...
```
Comment (kshakir) [inline build.sbt]: `inConfig(DbmsTest)` (2 of 2)

**Comment by kshakir** [comment]
After the changes above, I believe the various Specs should run as expected:

- `sbt alltests:test`
- `sbt notests:test`
- `sbt nodocker:test`
- `sbt dbms:test`
- `sbt integration:test`
- etc.

File: project/Testing.scala (hunk 2 above)
Comment (cjllanwarne) [inline project/Testing.scala]: I think @kshakir has FTFM. Thanks!

**Comment by kshakir** [comment]
:+1:

**Comment by scottfrazer** [comment]
:+1:

**Pull request merged**